trieste/models/optimizer.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, Optional, Callable, Union, Iterable, Tuple
+from functools import singledispatch
+from dataclasses import dataclass, field
+import tensorflow as tf
+import gpflow
+
+from gpflow.models import ExternalDataTrainingLossMixin, InternalDataTrainingLossMixin
+from ..utils import jit
+from ..datasets import Dataset
+
+
+DatasetBuilder = Callable[[Dataset, Optional[int]], Union[Iterable, Tuple]]
```

Comment:
I would expect a `DatasetBuilder` to build datasets. Was there something wrong with the name `Batcher`?

Reply:
It is not necessary a batcher, it can be a data transformer.

Reply:
ok, but `DatasetBuilder` gives the wrong impression, as it doesn't make `Dataset`s

Reply:
DatasetTransformer